src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Summarize.cs

```diff
@@ -0,0 +1,183 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using Microsoft.PowerFx.Core.App.ErrorContainers;
+using Microsoft.PowerFx.Core.Errors;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Syntax;
+
+namespace Microsoft.PowerFx.Core.Texl.Builtins
+{
+    // Summarize( Table, GroupByColumn1 [, GroupByColumn2 …], AggregateExpr1 As Name [, AggregateExpr1 As Name …] )
+    internal sealed class SummarizeFunction : FunctionWithTableInput
+    {
+        public override bool IsSelfContained => true;
+
+        public override bool SupportsParamCoercion => false;
+
+        public override bool HasColumnIdentifiers => true;
+
+        private static readonly IReadOnlyCollection<string> _supportedAggregateFunctions = new HashSet<string>()
```

Comment:
>_supportedAggregateFunctions [](http://example.com/codeflow?start=60&length=28)

where do we use this?

Reply:
On second thought, I removed this for now since the class already validates any aggregation class that returns anything different than an primitive result.